src/main/java/io/leonis/torch/component/TextBackground.java

```diff
@@ -0,0 +1,60 @@
+package io.leonis.torch.component;
+
+import com.googlecode.lanterna.*;
+import com.googlecode.lanterna.gui2.*;
+import java.awt.Color;
+
+/**
+ * The Class TextBackground
+ *
+ * This class is basically an {@link EmptySpace} with a text on top. The text is a tint brighter
+ * than the background color and will be centered based on the current size of the {@link
+ * com.googlecode.lanterna.screen.Screen}
+ *
+ * @author Jeroen de Jong
+ */
+public final class TextBackground extends EmptySpace {
+
+  /**
+   * Background label
+   */
+  private final Label text;
+
+  /**
+   * Creates a new TextBackground
+   *
+   * @param background Color for the background
+   * @param text Text to display on top of the background.
+   */
+  public TextBackground(final TextColor background, final String text) {
+    super(background);
+    this.text = new Label(text);
+  }
+
+  @Override
+  protected ComponentRenderer<EmptySpace> createDefaultRenderer() {
```

Comment:
how often is this method called? it might be better to provide the below anonymous class as a static inner class in order to prevent dynamically extending ComponentRenderer with a new anonymous type every time the function is called

Reply:
Once per render call; but it's all cached internally afaik. It's usually done this way, [see lanterna](https://github.com/mabe02/lanterna/blob/master/src/main/java/com/googlecode/lanterna/gui2/EmptySpace.java#L87)